lib/components/Account/AccountMenuButton.tsx

```diff
@@ -0,0 +1,130 @@
+import { EnterIcon, XMarkIcon } from '@navikt/aksel-icons';
+import cx from 'classnames';
+import { isValidElement } from 'react';
+import { type AccountMenuItemProps, ButtonBase, ButtonIcon, ButtonLabel, type ButtonProps } from '..';
+import { Avatar, type AvatarProps } from '../Avatar';
+
+import styles from './accountMenuButton.module.css';
+
+export interface AccountMenuButtonProps extends ButtonProps {
+  currentAccount?: AccountMenuItemProps;
+  className?: string;
+  expanded?: boolean;
+  /** Display the button as a minimized, mobile-friendly version */
+  minimized?: boolean;
+  tabIndex?: number;
+  loading?: boolean;
+}
+
+export const AccountMenuButton = ({
+  className,
+  currentAccount,
+  expanded,
+  minimized = false,
+  loading = false,
+  ...buttonProps
+}: AccountMenuButtonProps) => {
+  const texts = getTexts(undefined);
+
+  if (loading) {
+    return (
+      <ButtonBase
+        {...buttonProps}
+        as={'button'}
+        type="button"
+        variant={'text'}
+        color={'accent'}
+        className={cx(styles.button, styles.loading, className)}
+        disabled
+        aria-label="loading"
+      >
+        <ButtonIcon className={styles.avatarIcon} icon={<Avatar name="loading" loading className={styles.avatar} />} />
+      </ButtonBase>
+    );
+  }
+
+  if (currentAccount) {
+    let description = currentAccount.description;
+    if (currentAccount.type === 'subunit' && typeof description === 'string' && description) {
+      const orgNoDescription = description.split(',');
+      description = `${orgNoDescription[0]}, ${texts.subunit}`;
+    }
+    return (
+      <ButtonBase
+        {...buttonProps}
+        as={'button'}
+        type="button"
+        variant={'text'}
+        color={'accent'}
+        className={cx(styles.button, className)}
+        aria-label={expanded ? texts.close : currentAccount.name}
+      >
+        {!minimized && (
+          <ButtonLabel>
+            {
+              <div className={styles.labelContainer}>
+                {currentAccount.name}
+                <span className={styles.description}>
+                  {typeof description === 'string' ? description : isValidElement(description) ? description : null}
+                </span>
+              </div>
+            }
+          </ButtonLabel>
+        )}
+        {expanded ? (
+          <ButtonIcon className={styles.closeIcon} icon={<XMarkIcon className={styles.icon} aria-hidden />} />
+        ) : (
+          <ButtonIcon
+            className={styles.avatarIcon}
+            icon={<Avatar {...(currentAccount?.icon as AvatarProps)} className={styles.avatar} />}
```

Comment:
ButtonIcon takes avatarProps directly, so no need to use Avatar :)

Reply:
@allinox but `size` is a prop?

Reply:
And overriding avatar with CSS is a hack, imho. It creates inconsistencies in the design.

Reply:
Also, I'm not sure that the sizing prop of the ButtonIcon actually works as intended, but I'm not going to spend time looking too much into that in this PR (from what I can remember from the last time I looked at the Avatar component, the sizing is pretty tied to the Designsystem sizing system, which is already outdated and should be updated,, but not in this PR)